core/jobs/types/model_property_test.py

```diff
@@ -49,96 +54,110 @@ class RepeatedValueModel(base_models.BaseModel):
 
 class ModelPropertyTests(test_utils.TestBase):
 
-    def setUp(self):
+    def setUp(self) -> None:
         self.id_property = model_property.ModelProperty(
             SubclassOfBaseModel, SubclassOfBaseModel.id)
         self.ndb_property = model_property.ModelProperty(
             SubclassOfBaseModel, SubclassOfBaseModel.value)
         self.ndb_repeated_property = model_property.ModelProperty(
             RepeatedValueModel, RepeatedValueModel.values)
 
-    def test_init_with_id_property(self):
+    def test_init_with_id_property(self) -> None:
         # Does not raise.
         model_property.ModelProperty(
             SubclassOfBaseModel, SubclassOfBaseModel.id)
 
-    def test_init_with_ndb_property(self):
+    def test_init_with_ndb_property(self) -> None:
         # Does not raise.
         model_property.ModelProperty(
             SubclassOfBaseModel, SubclassOfBaseModel.value)
 
-    def test_init_with_ndb_repeated_property(self):
+    def test_init_with_ndb_repeated_property(self) -> None:
         # Does not raise.
         model_property.ModelProperty(
             RepeatedValueModel, RepeatedValueModel.values)
 
-    def test_init_raises_type_error_when_model_is_not_a_class(self):
+    def test_init_raises_type_error_when_model_is_not_a_class(self) -> None:
         model = SubclassOfBaseModel()
 
-        with self.assertRaisesRegex(TypeError, 'not a model class'):
-            model_property.ModelProperty(model, SubclassOfBaseModel.value)
-
-    def test_init_raises_type_error_when_model_is_unrelated_to_base_model(self):
-        with self.assertRaisesRegex(TypeError, 'not a subclass of BaseModel'):
+        with self.assertRaisesRegex(TypeError, 'not a model class'):  # type: ignore[no-untyped-call]
+            # ModelProperty has model_class argument, which can only accept
+            # classes that are inherited from BaseModel. But here we are
+            # passing an object of SubclassOfBaseModel. So, to avoid mypy
+            # error we added an ignore here.
+            model_property.ModelProperty(model, SubclassOfBaseModel.value)  # type: ignore[arg-type]
```

Comment:
Should the `model_property.ModelProperty` be retyped so that it can accept the subclass and also the class below?

Reply:
I think the constructor of `model_property.ModelProperty` is already accepting subclasses of `base_models.BaseModel`.
https://github.com/oppia/oppia/blob/ed2f6993a118527e169dccc4b44a39b81bebdf63/core/jobs/types/model_property.py#L50

But in this test we are passing object not class that's why comment and ignore is added, and in the test below we are passing subclass of `datastore_services.Model` not the class of `base_models.BaseModel` that's why comment and ignore is added there also.